src/tests/test_utils/common_test_utils/src/ov_plugin_cache.cpp

```diff
@@ -15,6 +15,61 @@
 namespace ov {
 namespace test {
 namespace utils {
+
+ov::AnyMap pluginConfig = {};
+
+ov::Core create_core(const std::string& target_device) {
+    ov::Core ov_core;
+
+    if (!pluginConfig.empty()) {
+        if (!target_device.empty()) {
+            auto properties = ov_core.get_property(target_device, ov::supported_properties);
+            for (auto& property : pluginConfig) {
+                if (std::find(properties.begin(), properties.end(), property.first) == properties.end()) {
+                    OPENVINO_THROW("Property " + property.first +
+                                   ", which was tryed to set in --config file, is not supported by " + target_device);
+                }
+            }
+            ov_core.set_property(target_device, pluginConfig);
+        } else {
+            ov_core.set_property(pluginConfig);
+        }
+    }
+
+    // Register Template plugin as a reference provider
+    // const auto devices = ov_core.get_available_devices();
+    try {
+        ov_core.get_property(ov::test::utils::DEVICE_TEMPLATE, ov::supported_properties);
+    } catch (const std::exception& ex) {
+        auto plugin_path =
+            ov::util::make_plugin_library_name(ov::test::utils::getExecutableDirectory(),
+                                               std::string(ov::test::utils::TEMPLATE_LIB) + OV_BUILD_POSTFIX);
+        if (!ov::util::file_exists(plugin_path)) {
+            OPENVINO_THROW("Plugin: " + plugin_path + " does not exists!");
+        }
+        ov_core.register_plugin(plugin_path, ov::test::utils::DEVICE_TEMPLATE);
+    }
+
+    if (!target_device.empty()) {
+        auto properties = ov_core.get_property(target_device, ov::supported_properties);
+
+        if (std::find(properties.begin(), properties.end(), ov::available_devices) != properties.end()) {
+            const auto available_devices = ov_core.get_property(target_device, ov::available_devices);
```

Comment:
Let's define `available_devices` as static class field to avoid of multiple getting of property